core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -265,7 +265,7 @@ public void print(TaskListener listener) {
 
         @Override
         public boolean equals(Object o) {
-            return o instanceof CLICause;
+            return o != null && o.getClass() == getClass();
```

Comment:
@StefanSpieker so the previous equals was not correct and behaviour changes now?

Reply:
The old version is not incorrect, but the new onemakes [the spotbugs rule](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#eq-equals-method-fails-for-subtypes-eq-getclass-and-class-constant) happy. The only diffference is that they behave differently for subclasses and it seems there are no subclasses of CLICause other than mocks: https://github.com/search?q=extends+CLICause+user%3Ajenkinsci&type=Code&ref=advsearch&l=&l=
So if [the tests using the mock](https://github.com/search?q=mock+CLICause+user%3Ajenkinsci&type=Code) are passing, all  should be fine.

Reply:
The fix looks good to me but I'm a bit concerned here. Why should this object be equal to another one just by being the same class? Shouldn't we check the `startedBy` field?

@oleg-nenashev do you have any idea? It seems to be a bug